scripts/jinja_model_gen.py

```diff
@@ -0,0 +1,169 @@
+#!/usr/bin/env python3
```

Comment:
This file looks like a duplicate of `jinja_gen.py`. Why was this file added?

Reply:
I'm leaving jinja_gen.py for now but it will be deleted/deprecated in the future, this allows for generation of models even from json source as well as checking for model options etc.

Reply:
@bperseghetti let's make this more consistent then. If we are replacing something, let's remove the part being replaced and not leave there as a remaining.

Reply:
I plan to give a one month phase out to cover any potential unforeseen issues and soft-add any extra desired features before eliminating the other pipeline.

Reply:
@bperseghetti I think these changes should be atomic given that they need to be part of the CI.

Reply:
So would you like me to then add this one instead into CMake to use for default builds?

Reply:
Yes, or make this `jinja_gen.py`? End of the day, it is functionally it is completely redundant right?

Reply:
Well, the reason why I left jinja_gen.py in place for now was this current implementation needs to be changed to allow for default generation from CMake or CMake needs to be modified to use it properly. So they are not actually "redundant". But I'll change CMake I guess if we want to eliminate the other script completely.

Reply:
If it is not redundant, we can leave it - but I didn't quite understand what you mean by not completely redundant. What is different? As far as I can see, they are both used to generate models from the same template?